packages/vite/src/client/overlay.ts

```diff
@@ -7,9 +7,36 @@ declare const __HMR_CONFIG_NAME__: string
 const hmrConfigName = __HMR_CONFIG_NAME__
 const base = __BASE__ || '/'
 
-// set :host styles to make playwright detect the element as visible
-const template = /*html*/ `
-<style>
+// Create an element with provided attributes and optional children
+function h(
+  e: string,
+  attrs: Record<string, string> = {},
+  ...children: (string | Node)[]
+) {
+  const elem = document.createElement(e)
+  for (const [k, v] of Object.entries(attrs)) {
+    if (k === 'class') {
+      elem.className = v
+    } else {
+      elem.setAttribute(k, v)
+    }
```

Comment:
```suggestion
    elem.setAttribute(k, v)
```
Is this condition necessary? If it's for compat with old browsers, let's remove it because Vite only supports latest browsers in dev.

Reply:
Turns out, it's not. While you can set `class` on an element, it doesn't actually equate to setting the class attribute to what you want. You either need to set `className` or take the input from `class`, split it at the spaces, then spread it over `classList.add`

Reply:
> While you can set class on an element, it doesn't actually equate to setting the class attribute to what you want.

What is the specific difference between `setAttribute('class', foo)` and `className = foo`?

Reply:
Ya know what? I read this wrong, I read it as setting the attribute directly because of other work I had done to get to here, but isn't actually needed. Yah, this is fine.